app/validators/ec_public_key_validator.rb

```diff
@@ -0,0 +1,15 @@
+class EcPublicKeyValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    regex = /\A-----BEGIN PUBLIC KEY-----.+-----END PUBLIC KEY-----\Z/m
+
+    begin
+      OpenSSL::PKey::EC.new(value)
+
+      if value !~ regex
```

Comment:
maybe better to get rid of the regex and do:
`
key = OpenSSL::PKey::EC.new(value)

if key.public?
`
?